src/content_scripts/autocorrect.js

```diff
@@ -0,0 +1,400 @@
+    const count = countChars(atext);
+    if (count > 0) {
+        const isSuccess = document.execCommand("delete", false);
+        if (isSuccess) {
+            for (let i = 0; i < count - 1; ++i) {
+                document.execCommand("delete", false);
+            }
+        }
+        // Firefox input and textarea fields: https://bugzilla.mozilla.org/show_bug.cgi?id=1220696
+        else if (typeof target.setRangeText === "function") {
+            const start = target.selectionStart;
+
+            target.selectionStart = start - atext.length;
+            target.selectionEnd = start;
+            target.setRangeText("");
+
+            // Notify any possible listeners of the change
+            const e = document.createEvent("UIEvent");
+            e.initEvent("input", true, false);
+            target.dispatchEvent(e);
+        }
+    }
+}
+
+/**
+ * Convert fractions and constants to Unicode characters.
+ * Adapted from: https://github.com/tdulcet/Tables-and-Graphs/blob/master/graphs.hpp
+ *
+ * @param {number} anumber
+ * @param {number} afraction
+ * @returns {string}
+ */
+function outputLabel(anumber, afraction) {
+    let output = false;
+
+    const number = parseFloat(anumber);
+    let intpart = Math.trunc(number);
+    const fractionpart = afraction ? parseFloat(afraction) : Math.abs(number % 1);
+
+    let strm = "";
+
+    for (const fraction in fractions) {
+        if (Math.abs(fractionpart - fractions[fraction]) < Number.EPSILON) {
+            if (intpart !== 0) {
+                strm += intpart;
+            }
+
+            strm += fraction;
+
+            output = true;
+            break;
+        }
+    }
+
+    if (Math.abs(number) >= Number.EPSILON && !output) {
+        for (const constant in constants) {
+            if (!output && number % constants[constant] === 0) {
+                intpart = number / constants[constant];
+
+                if (intpart === -1) {
+                    strm += "-";
+                } else if (intpart !== 1) {
+                    strm += intpart;
+                }
+
+                strm += constant;
+
+                output = true;
+                break;
+            }
+        }
+    }
+
+    if (!output) {
+        strm += anumber;
+    }
+
+    return strm;
+}
+
+/**
+ * Get first difference index.
+ *
+ * @param {string} a
+ * @param {string} b
+ * @returns {number}
+ */
+function firstDifferenceIndex(a, b) {
+    if (a === b) {
+        return -1;
+    }
+    let i = 0;
+    while (a[i] === b[i]) {
+        ++i;
+    }
+    return i;
+}
+
+/**
+ * Autocorrect on text input even by evaluating the keys and replacing the characters/string.
+ *
+ * @param {Object} event
+ * @returns {void}
+ */
+function autocorrect(event) {
+    // console.log('keydown', event.key, event.key.length, event.keyCode);
+    if (!((event.key.length === 0 || event.key.length === 1 || event.keyCode === 13 || event.key === "Unidentified") && !event.ctrlKey && !event.metaKey && !event.altKey)) {
```

Comment:
This exactly does what?

E.g.:
```suggestion
    // exclude common, for our use case uninteresting, keys from being processed
    if (!((event.key.length === 0 || event.key.length === 1 || event.keyCode === 13 || event.key === "Unidentified") && !event.ctrlKey && !event.metaKey && !event.altKey)) {
```

Reply:
This is a bunch of hacks to exclude all keys that do not produce a single Unicode character. The `event.key.length === 1` handles most of these cases. The `event.keyCode === 13` is the for space bar. The `event.key.length === 0` and `event.key === "Unidentified"` are for unrecognized keys, which occurs a lot with onscreen keyboards, particularly on Android and with emoji pickers. You can uncomment the line above and start pressing keys on different keyboards to see how they register. I will try to add a comment here.